library/std/src/sys/fs/windows.rs

```diff
@@ -846,6 +851,217 @@ impl File {
     }
 }
 
+unsafe fn nt_create_file(
+    access: u32,
+    object_attributes: &c::OBJECT_ATTRIBUTES,
+    share: u32,
+    dir: bool,
+) -> Result<Handle, WinError> {
+    let mut handle = ptr::null_mut();
+    let mut io_status = c::IO_STATUS_BLOCK::PENDING;
+    let disposition = match (access & c::GENERIC_READ > 0, access & c::GENERIC_WRITE > 0) {
+        (true, true) => c::FILE_OPEN_IF,
+        (true, false) => c::FILE_OPEN,
+        (false, true) => c::FILE_CREATE,
+        (false, false) => {
+            return Err(WinError::new(c::ERROR_INVALID_PARAMETER));
+        }
+    };
```

Comment:
This needs at least a few extra access and create options.

```suggestion
    };
    let access = access | c::SYNCHRONIZE;
    let options = if dir { c::FILE_DIRECTORY_FILE } else { c::FILE_NON_DIRECTORY_FILE }
        | c::FILE_SYNCHRONOUS_IO_NONALERT;
```